parquet-hadoop/src/test/java/org/apache/parquet/statistics/RandomValueGenerators.java

```diff
@@ -0,0 +1,335 @@
+    }
+  }
+
+  public static class RandomInt96Generator extends RandomValueGenerator<BigInteger> {
+    public RandomInt96Generator(long seed, int total, BigInteger minimum, BigInteger maximum) {
+      super(seed, PrimitiveType.PrimitiveTypeName.INT96, total, minimum, maximum);
+    }
+
+    @Override
+    protected List<BigInteger> generateValues(int total, BigInteger minimum, BigInteger maxiumum) {
+      BigInteger range = maxiumum.subtract(minimum);
+      List<BigInteger> values = new ArrayList<BigInteger>();
+
+      for (long index = 0; index < total; index++) {
+        if (!shouldGenerateNull()) {
+          values.add(minimum.add(randomInt96(range, super.random)));
+        } else {
+          values.add(null);
+        }
+      }
+
+      return values;
+    }
+
+    public static Binary asBinary(BigInteger value) {
+      byte[] asBytes = Arrays.copyOf(value.toByteArray(), 12);
+      return Binary.fromConstantByteArray(asBytes);
+    }
+
+    public Binary asBinary(int index) {
+      BigInteger value = values.get(index);
+
+      if (value != null) {
+        return asBinary(value);
+      } else {
+        return null;
+      }
+    }
+  }
+
+
+  public static class RandomFloatGenerator extends RandomValueGenerator<Float> {
+    public RandomFloatGenerator(long seed, int total, float minimum, float maximum) {
+      super(seed, PrimitiveType.PrimitiveTypeName.FLOAT, total, minimum, maximum);
+    }
+
+    @Override
+    protected List<Float> generateValues(int total, Float minimum, Float maxiumum) {
+      float range = (maxiumum - minimum);
+      List<Float> values = new ArrayList<Float>();
+
+      for (long index = 0; index < total; index++) {
+        if (!shouldGenerateNull()) {
+          values.add(minimum + randomFloat(range));
+        } else {
+          values.add(null);
+        }
+      }
+
+      return values;
+    }
+  }
+
+  public static class RandomDoubleGenerator extends RandomValueGenerator<Double> {
+    public RandomDoubleGenerator(long seed, int total, double minimum, double maximum) {
+      super(seed, PrimitiveType.PrimitiveTypeName.DOUBLE, total, minimum, maximum);
+    }
+
+    @Override
+    protected List<Double> generateValues(int total, Double minimum, Double maxiumum) {
+      double range = (maxiumum - minimum);
+      List<Double> values = new ArrayList<Double>();
+
+      for (long index = 0; index < total; index++) {
+        if (!shouldGenerateNull()) {
+          values.add(minimum + randomDouble(range));
+        } else {
+          values.add(null);
+        }
+      }
+
+      return values;
+    }
+  }
+
+  public static class RandomBinaryGenerator extends RandomValueGenerator<String> {
+    public RandomBinaryGenerator(long seed, int total) {
+      super(seed, PrimitiveType.PrimitiveTypeName.BINARY, total);
+    }
+
+    @Override
+    protected List<String> generateValues(int total, String minimum, String maxiumum) {
+      List<String> values = new ArrayList<String>();
```

Comment:
If you seed the random generator with the same values (or add a reset method) then you shouldn't need to keep the values around for the test. That would decrease the chances of hitting an OOM.